### PR TITLE
fix: Refactor init mongodb flow to check and enable replica set for external mongodb 

### PR DESCRIPTION
**Pull request by geekup-legodevops**
## Description
- Add flow to check if MongoDB URI pointing to local service or not. Perform local mongodb init as normal
- Remove unnecessary environment variables for MongoDB credentials and directly extract credentials from URI
- Add prefix /api/v1 in the API_BASE_URL 
- Add flow to check external MongoDB is set as replica set or not. If yes, initiate the replica set. If not, print out error message and stop installation.

Fixes # (issue)

[[Bug]: Using external MongoDB will still try to connect and configure local MongoDB](https://github.com/appsmithorg/appsmith/issues/9513)

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: deploy/docker/entrypoint.sh [hunk 1]
```diff
@@ -2,7 +2,35 @@
...
 set -e
 
-check_initialized_db() {
+get_variable_mongodb() {
+  echo "Get variable mongodb"
+  MONGODB_PROTOCOL=$(echo "$APPSMITH_MONGODB_URI" | grep -oP "[a-z+]+(?=\:\/\/)") 
+  MONGODB_USERNAME=$(echo "$APPSMITH_MONGODB_URI" | grep -oP "(?<=\:\/\/)(.*\s?)(?=\:)")
+  MONGODB_PASSWORD=$(echo "$APPSMITH_MONGODB_URI" | grep -oP "\w+(?=@)")
+  MONGODB_HOST=$(echo "$APPSMITH_MONGODB_URI" | grep -oP "[a-z0-9.]+(?=\/)")
+  MONGODB_DATABASE=$(echo "$APPSMITH_MONGODB_URI" | grep -oP "\w+(?!\S)")
```
Comment (sharat87) [inline deploy/docker/entrypoint.sh:11]: Can you confirm if this works when the URI has query params with `?authSource=admin` as well as when there are no query params at all?

File: deploy/docker/entrypoint.sh [hunk 2]
```diff
@@ -12,42 +40,47 @@ check_initialized_db() {
...
     sleep 10
     mongo "$APPSMITH_MONGODB_URI" --eval 'rs.initiate()'
     mongod --dbpath "$MONGO_DB_PATH" --shutdown || true
   fi
+
+  if [[ $isUriLocal -gt 0 ]]; then
+    # Check mongodb cloud replica set
+    echo 'Check mongodb cloud replica set'
+    responseStatus=$(mongo "$APPSMITH_MONGODB_URI" --eval "rs.status()" | grep ok | xargs)
+    okString='ok : 1'
```
Comment (sharat87) [inline deploy/docker/entrypoint.sh:72]: I just ran this command and the output shows `ok: 1`, not `ok : 1`, is that okay? Will it still match?

File: deploy/docker/entrypoint.sh [hunk 3]
```diff
@@ -67,7 +100,7 @@ configure_supervisord() {
   cp -f "$SUPERVISORD_CONF_PATH/application_process/"*.conf /etc/supervisor/conf.d
 
   # Disable services based on configuration
-  if [[ "$APPSMITH_MONGODB_URI" = "mongodb://appsmith:$MONGO_INITDB_ROOT_PASSWORD@localhost/appsmith" ]]; then
+  if [[ "$APPSMITH_MONGODB_URI" = "mongodb://appsmith:$MONGODB_PASSWORD@$MONGODB_HOST/appsmith" && $isUriLocal -eq 0 ]]; then
```
Comment (sharat87) [inline deploy/docker/entrypoint.sh:103]: Isn't checking `isUriLocal` enough here? In the first check, we are basically checking is username is `appsmith` and database name is `appsmith`. The other two things, password and host, are basically extracted from `APPSMITH_MONGODB_URI` only, so they'll anyway match.

**Review by sharat87 — CHANGES_REQUESTED** [review]
In addition to these comments, I _think_ there's a conflict as well. Please merge release back into this branch just in case.

File: deploy/docker/entrypoint.sh (hunk 2 above)
Comment (geekup-legodevops) [reply deploy/docker/entrypoint.sh:72]: Hi @sharat87 have you run the whole command with the 'xargs' command ? Last time we checked, the 'xargs' command will make the `ok: 1` string to `ok : 1`

Comment (sharat87) [reply deploy/docker/entrypoint.sh:72]: Oh I see, cool, thanks for confirming.

**Review by sharat87 — APPROVED** [review]

**Comment by MichaelLeeHobbs** [comment]
#11418 Bug on line 125

```
  if [[ $isUriLocal -gt 0 ]]; then
```

Should be

```
  if [[ $isUriLocal -eq 0 ]]; then
```

**Pull request merged**